libgit/autogit_manager.go

```diff
@@ -225,10 +282,84 @@ func (am *AutogitManager) TlfHandleChange(
 	// Do nothing.
 }
 
+// GetBrowserForRepo returns the root FS for the specified repo and a
+// `Browser` for the branch and subdir.
+func (am *AutogitManager) GetBrowserForRepo(
+	ctx context.Context, gitFS *libfs.FS, repoName string,
+	branch plumbing.ReferenceName, subdir string) (*libfs.FS, *Browser, error) {
+	key := browserCacheKey{gitFS, repoName, branch, subdir}
+
+	am.browserLock.Lock()
+	defer am.browserLock.Unlock()
+	tmp, ok := am.browserCache.Get(key)
+	if ok {
+		b, ok := tmp.(browserCacheValue)
+		if !ok {
+			return nil, nil, errors.Errorf("Bad browser in cache: %T", tmp)
+		}
+		return b.repoFS, b.browser, nil
+	}
+
+	// It's kind of dumb to hold the browser lock through all of this,
+	// but it doesn't seem worthwhile to build the whole
+	// channel/notification system that would be needed to manage
+	// multiple concurrent requests for the same repo node.
+
+	am.log.CDebugf(ctx, "Making browser for repo=%s, branch=%s, subdir=%s",
+		repoName, branch, subdir)
+
```

Comment:
I wonder if we can recurse into calling `am.GetBrowserForRepo(ctx, gitFS, repoName, branch, "")` here, Chroot, then `Add` the subdir one. This way the part below (during cache miss) can be simplified too.

Reply:
Great idea!  Done in e730dd3.